examples/pytorch/image-classification/run_image_classification.py

```diff
@@ -201,25 +201,44 @@ def main():
             )
 
     # Initialize our dataset and prepare it for the 'image-classification' task.
-    ds = load_dataset(
-        data_args.dataset_name,
-        data_args.dataset_config_name,
-        data_files=data_args.data_files,
-        cache_dir=model_args.cache_dir,
-        task="image-classification",
-        use_auth_token=True if model_args.use_auth_token else None,
-    )
+    if data_args.dataset_name is not None:
+        dataset = load_dataset(
+            data_args.dataset_name,
+            data_args.dataset_config_name,
+            cache_dir=model_args.cache_dir,
+            task="image-classification",
+            use_auth_token=True if model_args.use_auth_token else None,
+        )
+    else:
+        dataset_dict = {}
+        if data_args.train_dir is not None:
+            dataset_dict["train"] = load_dataset(
+                "imagefolder",
+                data_dir=data_args.train_dir,
+                cache_dir=model_args.cache_dir,
+                task="image-classification",
+                split="train",
+            )
+        if data_args.validation_dir is not None:
+            dataset_dict["validation"] = load_dataset(
+                "imagefolder",
+                data_dir=data_args.validation_dir,
+                cache_dir=model_args.cache_dir,
+                task="image-classification",
+                split="validation",
+            )
+        dataset = DatasetDict(dataset_dict)
```

Comment:
cc @nateraw @mariosasko , this is how I'm loading local folders now.

Reply:
ImageFolder + `data_dir` collects everything in a `train` split, so this part is problematic:
```python
 dataset_dict["validation"] = load_dataset(
                "imagefolder",
                data_dir=data_args.validation_dir,
                cache_dir=model_args.cache_dir,
                task="image-classification",
                split="validation", <-
            )
```

We can rewrite this part as follows to make it cleaner and more robust (for instance, this approach doesn't fail if some labels are not present in the validation `data_dir`):
```python
data_files = {}
if data_args.train_dir is not None:
    data_files["train"] = os.path.join(data_args.train_dir, "**")
if data_args.validation_dir is not None:
    data_files["validation"] = os.path.join(data_args.validation_dir, "**")
dataset = load_dataset(
    "imagefolder",
     data_files=data_files,
     cache_dir=model_args.cache_dir,
     task="image-classification",
)
```